### PR TITLE
Move data files out of test directory

### PR DESCRIPTION
**Pull request by spxiwh**
Responding to #4104 this removes the `test/data` directory. Data files are now moved to the `pycbc-config` repository and pulled from there using astropy. This limits the filesize of PyCBC distributions, while still enabling the full test suite to be run when needed.

### CODE REVIEW

File: test/test_skymax.py [hunk 1]
```diff
@@ -305,10 +308,22 @@ def test_filtering(self):
         #    hcross.save('data/skymaxtest_hcross_%d.hdf' % jdx)
         #except:
         #    pass
-        hplus = load_frequencyseries(self.dataDir + \
-                                     'skymaxtest_hplus_%d.hdf' % jdx)
-        hcross = load_frequencyseries(self.dataDir + \
-                                      'skymaxtest_hcross_%d.hdf' % jdx)
+        fname = 'skymaxtest_hplus_{}.hdf'.format(jdx)
```
Comment (titodalcanton) [inline test/test_skymax.py:311]:
```suggestion
        fname = f'skymaxtest_hplus_{jdx}.hdf'
```

File: test/test_skymax.py [hunk 2]
```diff
@@ -305,10 +308,22 @@ def test_filtering(self):
...
+        apy_fname = download_file(url.format(fname), cache=False)
+        # Astropy will not download with the .hdf extension, which we need,
+        # so symlink
+        os.symlink(apy_fname, fname)
+        hplus = load_frequencyseries(fname)
+        os.unlink(fname)
+
+        fname = 'skymaxtest_hcross_{}.hdf'.format(jdx)
```
Comment (titodalcanton) [inline test/test_skymax.py:319]:
```suggestion
        fname = f'skymaxtest_hcross_{jdx}.hdf'
```

File: test/test_skymax.py [hunk 3]
```diff
@@ -291,8 +286,16 @@ def test_filtering(self):
         #    stilde.save('data/skymaxtest_stilde_%d.hdf' % idx)
         #except:
         #    pass
-        stilde = load_frequencyseries(self.dataDir + \
-                                      'skymaxtest_stilde_%d.hdf' % idx)
+        url = 'https://github.com/gwastro/pycbc-config/raw/master/'
+        url += 'test_data_files/{}'
+        fname = 'skymaxtest_stilde_{}.hdf'.format(idx)
```
Comment (titodalcanton) [inline test/test_skymax.py:291]:
```suggestion
        url = ('https://github.com/gwastro/pycbc-config/raw/master/'
               'test_data_files/{}')
        fname = f'skymaxtest_stilde_{idx}.hdf'
```

File: test/test_tmpltbank.py [hunk 4]
```diff
@@ -196,13 +197,19 @@ def setUp(self):
...
+        apy_fname = download_file(DATA_FILE_URL.format(fname), cache=False)
+        evecsStock = Array(numpy.loadtxt(apy_fname))
+
         maxEval = max(evalsStock)
         evalsCurr = Array(self.metricParams.evals[self.f_upper])
         evecsCurr = Array(self.metricParams.evecs[self.f_upper])
-        numpy.savetxt('newEvals.dat', evalsCurr)
-        numpy.savetxt('newEvecs.dat', evecsCurr)
+        #numpy.savetxt('newEvals.dat', evalsCurr)
+        #numpy.savetxt('newEvecs.dat', evecsCurr)
```
Comment (titodalcanton) [inline test/test_tmpltbank.py:212]: I would just remove these lines rather than commenting them (same comment below).

**Review by ahnitz — APPROVED** [review]
@spxiwh I agree with @titodalcanton's comment to just remove the commented lines, but otherwise, this looks fine.

File: test/test_tmpltbank.py (hunk 4 above)
Comment (spxiwh) [reply test/test_tmpltbank.py:212]: ... I note that they should never have been uncommented in the version committed to the main branch. I should have commented them out before uploading the code in the first place.

**Comment by spxiwh** [comment]
I've added comments to explain the commented out blocks of code. Any objections to merging this now?

**Pull request merged**